### PR TITLE
Change the strip feature of the CPU template helper tool to operate bitwise

### PR DESCRIPTION
**Pull request by zulinx86**
## Changes

- Change the strip feature of the CPU template helper tool to operate bitwise.

## Reason

In a case where only 1 bit of a CPUID register value is different across inputs, all the other 31 bits are also preserved in the stripped outputs.
Ignoring all the other bits by eyes is quite troublesome from user perspective.

example of value-wise output
```
    {
      "leaf": "0x0",
      "subleaf": "0x0",
      "flags": 0,
      "modifiers": [
        {
          "register": "eax",
          "bitmap": "0b00000000000000000000000000001001"
        }
      ]
    },
```

example of bit-wise output
```
    {
      "leaf": "0x0",
      "subleaf": "0x0",
      "flags": 0,
      "modifiers": [
        {
          "register": "eax",
          "bitmap": "0bxxxxxxxxxxxxxxxxxxxxxxxxxxx01x01"
        }
      ]
    },
```

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- ~~[ ] If a specific issue led to this PR, this PR closes the issue.~~
- [x] The description of changes is clear and encompassing.
- ~~[ ] Any required documentation changes (code and docs) are included in this PR.~~
- ~~[ ] API changes follow the [Runbook for Firecracker API changes][2].~~
- [x] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.
- ~~[ ] New `TODO`s link to an issue.~~
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,8 @@
   custom CPU templates.
 - Added support for the `virtio-rng` entropy device. The device is optional. A
   single device can be enabled per VM using the `/entropy` endpoint.
+- Added a new CLI option `--bitwise` for the strip feature of
```
Comment (kalyazin) [inline CHANGELOG.md:15]: what could be a reason for users to opt out?

Comment (zulinx86) [reply CHANGELOG.md:15]: The reason is only that some bits are grouped. For example, when a group of 4 bits is 0b1100 and 0b1010, the output is going to be 0bx10x and 0x01x. In this case, users need to look at the input file with the output file to know the 'x' bits.

thb, I guess having only the bitwise one is fine. I'd appreciate it if you could share your thought.

Comment (zulinx86) [reply CHANGELOG.md:15]: Indeed. I don't think users want to think much about the usage of the tool, so it's better to keep it simple. Thank you for sharing your thought.

Actually what I implemented before publishing this PR was replace it with bit-wise one. I'll make this PR back.

Comment (zulinx86) [reply CHANGELOG.md:15]: Changed the opt-in approach to the approach to only have the bitwise strip.

File: src/cpu-template-helper/src/template/strip/mod.rs [hunk 2]
```diff
@@ -33,14 +33,50 @@ where
...
+            common_vf.filter = diff;
+        }
+    }
+
+    // Remove the `common` items from all the `maps`.
+    for (key, common_vf) in common {
         for map in maps.iter_mut() {
-            map.remove(key);
+            if common_vf.filter == V::zero() {
+                // Remove the `key` if the filtered value is idential in all the `maps`.
```
Comment (kalyazin) [inline src/cpu-template-helper/src/template/strip/mod.rs:73]: nit: idential -> identical

Comment (zulinx86) [reply src/cpu-template-helper/src/template/strip/mod.rs]: thanks! fixed.

**Review by kalyazin — DISMISSED** [review]

**Comment by zulinx86** [comment]
force-pushed to resolve conflict of CHANGELOG.md

**Comment by zulinx86** [comment]
force-pushed to resolve conflict of Cargo.toml

**Review by JonathanWoollett-Light — APPROVED** [review]

**Review by kalyazin — APPROVED** [review]

**Pull request merged**